pkg/apis/build/v1alpha2/build_pod.go

```diff
@@ -410,11 +409,12 @@ func (b *Build) BuildPod(images BuildPodImages, secrets []corev1.Secret, taints
 									return nil
```

Comment:
I think in all three cases we want to set a process type if the process type is set.